apps/3000-home/pages/_document.js

```diff
@@ -0,0 +1,30 @@
+import Document, { Html, Head, Main, NextScript } from 'next/document'
+import {revalidate} from "@module-federation/nextjs-mf/utils";
+
+class MyDocument extends Document {
+  static async getInitialProps(ctx) {
+    const initialProps = await Document.getInitialProps(ctx);
+
+    ctx?.res?.on('finish', () => {
+      revalidate().then((shouldUpdate) => {
```

Comment:
I don't think so because loading the remotes is async itself. Depending on how users want to revalidation they may wish to hold the app, hot reloading then render.